util/crypto/src/secp/signature.rs

```diff
@@ -74,7 +65,7 @@ impl Signature {
                 return false;
             }
         };
-        self.v() <= 1 && h_r < N && h_r >= ONE && h_s < N && h_s >= ONE
+        self.v() < 4 && h_r < N && h_r >= ONE && h_s < N && h_s >= ONE
```

Comment:
why is this change needed?

Reply:
After checking the source code of secp256k1 and peter wullies's answer, it seems in practice the `v` is only considered either `0` or `1`.

I'll rollback this.

https://github.com/bitcoin-core/secp256k1/blob/544435fc90a5672d862e2a51f44c10251893b97d/src/ecdsa_impl.h#L298

https://bitcoin.stackexchange.com/questions/38351/ecdsa-v-r-s-what-is-v

Reply:
After checking the source code of secp256k1 and peter wullies's answer, it seems in practice the `v` is only considered either `0` or `1`.

I'll rollback this.

https://github.com/bitcoin-core/secp256k1/blob/544435fc90a5672d862e2a51f44c10251893b97d/src/ecdsa_impl.h#L298

https://bitcoin.stackexchange.com/questions/38351/ecdsa-v-r-s-what-is-v